### ISSUE TITLE
fs_load.c: memcp used on overlapping memory ranges

### ISSUE THREAD

**Issue by joachimmetz** [sleuthkit/sleuthkit#2807]
https://github.com/sleuthkit/sleuthkit/blame/develop/tsk/fs/fs_load.c#L35 is called from https://github.com/sleuthkit/sleuthkit/blame/develop/tsk/fs/fs_attr.c#L840 as callback by https://github.com/sleuthkit/sleuthkit/blame/develop/tsk/fs/ntfs_dent.cpp#L1053

From memcpy(3)

```
DESCRIPTION
       The memcpy() function copies n bytes from memory area src to memory area dest.  The memory areas must not overlap.  Use memmove(3) if the memory areas do overlap.
```

**The memory areas must not overlap.  Use memmove(3) if the memory areas do overlap.**

**Comment by joachimmetz** [comment]
Suggested changes in https://github.com/sleuthkit/sleuthkit/pull/2808 though it looks like this overlapping ranges issue might be masking another issue.

**Comment by joachimmetz** [comment]
Per https://github.com/sleuthkit/sleuthkit/blame/develop/tsk/fs/ntfs_dent.cpp#L1041 if `idxalloc_len == 0` then https://github.com/sleuthkit/sleuthkit/blame/develop/tsk/fs/ntfs_dent.cpp#L1047 `load_file.cur` and `load_file.base` are set  to NULL. But tsk_fs_load_file_action does not check for this condition.

**Comment by joachimmetz** [comment]
Updated https://github.com/sleuthkit/sleuthkit/pull/2808

**Comment by lfcnassif** [comment]
Maybe this should be closed after #2808 was merged.
